src/main/java/graphql/annotations/processor/retrievers/GraphQLFieldRetriever.java

```diff
@@ -104,28 +102,62 @@ public GraphQLFieldDefinition getField(Method method, ProcessingElementsContaine
         GraphQLFieldDefinition.Builder builder = newFieldDefinition();
         TypeFunction typeFunction = getTypeFunction(method, container);
         builder.name(new MethodNameBuilder(method).build());
-        GraphQLOutputType outputType = new MethodTypeBuilder(method, typeFunction, container).build();
+        GraphQLOutputType outputType = (GraphQLOutputType) new MethodTypeBuilder(method, typeFunction, container, false).build();
 
         boolean isConnection = ConnectionUtil.isConnection(method, outputType);
         if (isConnection) {
             outputType = getGraphQLConnection(method, outputType, container.getRelay(), container.getTypeRegistry());
         }
         builder.type(outputType);
         handleConnectionArgument(container, builder, isConnection);
-        List<GraphQLArgument> args = new ArgumentBuilder(method, typeFunction, graphQLInputObjectRetriever, builder, container, outputType).build();
+        List<GraphQLArgument> args = new ArgumentBuilder(method, typeFunction, builder, container, outputType).build();
         GraphQLFieldDefinition relayFieldDefinition = handleRelayArguments(method, container, builder, outputType, args);
         builder.description(new DescriptionBuilder(method).build())
                 .deprecate(new DeprecateBuilder(method).build())
                 .dataFetcher(new MethodDataFetcherBuilder(method, outputType, typeFunction, container, relayFieldDefinition, args, dataFetcherConstructor, isConnection).build());
         return new GraphQLFieldDefinitionWrapper(builder.build());
     }
 
+    public GraphQLFieldDefinition getField(Field field, ProcessingElementsContainer container) throws GraphQLAnnotationsException {
+        GraphQLFieldDefinition.Builder builder = newFieldDefinition();
+        builder.name(new FieldNameBuilder(field).build());
+        TypeFunction typeFunction = getTypeFunction(field, container);
+
+        graphql.schema.GraphQLType outputType = typeFunction.buildType(field.getType(), field.getAnnotatedType(), container);
+        boolean isConnection = ConnectionUtil.isConnection(field, outputType);
+        if (isConnection) {
+            outputType = getGraphQLConnection(field, outputType, container.getRelay(), container.getTypeRegistry());
+            builder.argument(container.getRelay().getConnectionFieldArguments());
+        }
+
+        builder.type((GraphQLOutputType) outputType).description(new DescriptionBuilder(field).build())
+                .deprecate(new DeprecateBuilder(field).build())
+                .dataFetcher(new FieldDataFetcherBuilder(field, dataFetcherConstructor, outputType, typeFunction, container, isConnection).build());
```

Comment:
The purpose of a builder is the ability to build the object in a nicer way than just give it all to the constructor